classes/class-blog-handler.php

```diff
@@ -84,21 +102,44 @@ public function init( array $data, $user = null ) {
 	 * @return int
 	 */
 	public function first_or_create_blog( $make_private = false ) {
+
+		$this->validate_blog_data();
+
+		$blog_id = $this->get_blog_id_if_exists();
+
+		if ( null === $blog_id ) {
+			$this->fix_blog_id_from_path();
+			$blog_id = $this->get_blog_id_if_exists();
+		}
+
+		if ( null === $blog_id ) {
+			$blog_id = $this->create_blog( $make_private );
+		}
+
+		$this->blog_id = $blog_id;
+
+		return $blog_id;
+	}
+
+	/**
+	 * Manage additional blog options when a blog is created or viewed
+	 *
+	 * @return string
+	 */
+	public function set_additional_blog_options( array $options_to_set ) {
+		foreach ( $options_to_set as $blog_option_key => $blog_option_value ) {
+			return update_blog_option( $this->blog_id, $blog_option_key, $blog_option_value );
+		}
+	}
+
+	protected function validate_blog_data() {
 		if (
 			null === $this->course_id || null === $this->course_title || null === $this->domain ||
 			null === $this->resource_link_id || null === $this->username || null === $this->source_id ||
 			null === $this->site_category
 		) {
 			wp_die( 'Blog_Handler: You must set all data before calling first_or_create_blog' );
 		}
-
-		$blog_id = $this->get_blog_id_if_exists();
```

Comment:
This line has been moved up, as part of the refactoring, so the outcome should be the same.